src/app/course/services/CoursesService.ts

```diff
@@ -0,0 +1,45 @@
+import {TCourse} from "../models/course";
+import CoursesList from './cources.json';
+
+export class CoursesService {
```

Comment:
You need to add a decorator `@Injectable` for all services

Reply:
Take a look into https://angular.io/api/core/Injectable 
providedIn: 'root', might be useful for AuthService